fhirpath.rst

```diff
@@ -0,0 +1,179 @@
+   * - ``FhirEvaluationContext.ElementResolver``
+     - A delegate that resolves an uri to an instance of FHIR data (``ITypedElement``). This callback is used by the FHIR specific method ``resolve()``.
+
+Note that ``FhirEvaluationContext`` is only used by the POCO extension methods for FhirPath, as it provides a property for setting the resolver.
+
+Best practices
+--------------
+Although it is seemingly easy to invoke FhirPath, there are a few details that are easy to get wrong.
+
+Start evaluation from the root
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To make the ``resolve()`` function work well (e.g. to resolve to entries in a Bundle or to a contained resource), the FhirPath engine needs
+to have "seen" all the resources while navigating through the data, which means you need to evaluate Bundles from their roots.
+
+.. code-block:: csharp
+
+  Bundle b = new() {...}
+
+  // The engine has worked from the root of the bundle down, so it knows how to resolve to other entries
+  var active = b.Select("Bundle.entry.ofType(Patient).organization.resolve()");
+
+  // The engine was started from the nested Patient node, so does not know how to find other entries.
+  var org = Bundle.entry.OfType<Patient>[0];
+  var active2 = org.Select("organization.resolve()");
+
+  // This is fine too, since the context is transferred from call to call.
+  var org2 = b.Select("Bundle.entry.ofType(Patient)");
+  var active3 = org2.Select("organization.resolve()");
+
+Use a context constructor which takes a resource to set ``%resource``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Although not many FhirPath statements use the ``%resource`` and ``%rootResource`` environment variables, they *do* get used, and the default constructors will
+make it easy for you to *not* set them (blame us for that). To make sure these variables work well, you should pass a sensible ``EvaluationContext`` to the
+FhirPath functions, even though they are optional:
+
+.. code-block:: csharp
+
+   Patient p = new() {...}
+   var hasName = p.IsTrue("Patient.name.exists()", new FhirEvaluationContext(p.ToScopedNode()));
+
+As you can see, we are passing in a new ``FhirEvaluationContext``, constructed with a reference to the root of the object. Additionally, the FhirPath engine needs
+its data to be a `ScopedNode`. This is a wrapper for ``ITypedElement`` that keeps track of parent nodes, contained nodes
+an entry nodes in a ``Bundle``, and does the heavy lifting for making ``resolve()`` work (see previous section).
+
+
+Set the Resolver property in the FhirEvaluationContext
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Finally, the engine needs you to supply a delegate when you want ``resolve()`` to be able to reach out to instances of Resources (via uri) that it cannot locate itself.
+The delegate you needs to supply takes a single string parameter (the uri), and returns an ``ITypedElement``. Just like in the previous section, it would be best
```

Comment:
```suggestion
The delegate you need to supply takes a single string parameter (the uri), and returns an ``ITypedElement``. Just like in the previous section, it would be best
```